backend/azcredentials/credentials.go

```diff
@@ -0,0 +1,5 @@
+package azcredentials
+
+type AzureCredentials interface {
```

Comment:
If this package only includes this interface think it makes more sense to move it together with AzureOptions

Reply:
I added the `AzureCredentials` interface into this package do define different layers of abstraction. `AzureCredentials` is a base interface for Azure credentials, and by itself is not related to `httpclient.Options`.

In case if we need to introduce other top level abstractions for Azure credentials, they may also go to `azcredentials` package.

`httpclient.Options` on the over hand has a configuration section for `AzureCredentials` and references it. `httpclient.AzureOptions` struct has no importance over than just a property bag for `httpclient.Options`, so it's part of the `httpclient` package.

My proposed package organization:

#### Grafana Plugin SDK

`"github.com/grafana/grafana-plugin-sdk-go/backend/azcredentials"`

* Top-level interface `AzureCredentials`.

#### Grafana Azure SDK

`"github.com/grafana/grafana-azure-sdk-go/azcredentialsimpl"`

* Azure credentials definitions for different authentication types `AzureManagedIdentityCredentials`, `AzureClientSecretCredentials`.

`"github.com/grafana/grafana-azure-sdk-go/azreader"`

* A helper reader for deserialization of `AzureCredentials` from the `JsonData` structure. Plugins may choose to use this reader or implement their own in case if their `JsonData` schema is different.

`"github.com/grafana/grafana-azure-sdk-go/aztokenprovider"`

* The token provider implementation (and middleware)

I'm open for ideas how to better organize the packages.

Reply:
> and by itself is not related to httpclient.Options

It's referenced in httpclient.Options that was why I suggested to move it together:
```go
// AzureOptions Azure authentication options.
type AzureOptions struct {
	Credentials azcredentials.AzureCredentials
	Scopes      []string
}
```